### PR TITLE
Added Prime Factor [Javascript]

### LINKED ISSUE #314: Prime Factor[Javascript]
#### This issue is about [Prime Factor](https://en.wikipedia.org/wiki/Prime_factor).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] I ensure that I'm not assigned to 2 or more issues.
- [x] Assign this issue to me (I want to work on this).
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by rashinanwani**
Fixes #314

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: prime_factor/prime_factor.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+function primeFactor (n) {
+	
+	for (let i = 2; i <= n; i++) {
+		if (n%i==0) {
```
Comment (Monal5031) [inline prime_factor/prime_factor.js]: Please make `==` as `===` and also add space around binary operators.

File: prime_factor/prime_factor.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+			console.log(i);
+			while(n%i==0) {
```
Comment (Monal5031) [inline prime_factor/prime_factor.js]: Same here, make `==` as `===` and add space around binary operator.

File: prime_factor/prime_factor.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        n=n/i;
+      }
+	  }
+	}
+}
+
+function main () {
+	let n=8;
+  console.log('Prime Factors are:');
+  for (let i = 2;i <= primeFactor(n);i++ ) {
```
Comment (Monal5031) [inline prime_factor/prime_factor.js]: there should space after semicolon, as `for (let i = 2; i <= primeFactor(n); i++)`

File: prime_factor/prime_factor.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+function primeFactor (n) {
+
```
Comment (Monal5031) [inline prime_factor/prime_factor.js]: Remove this blank line.

File: prime_factor/prime_factor.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+function primeFactor (n) {
+	
+	for (let i = 2; i <= n; i++) {
+		if (n%i==0) {
+			console.log(i);
+			while(n%i==0) {
+        n=n/i;
```
Comment (Monal5031) [inline prime_factor/prime_factor.js]: Add space around binary operators here too.

File: prime_factor/prime_factor.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+function primeFactor (n) {
+	for (let i = 2; i <= n; i++) {
```
Comment (aashutoshrathi) [inline prime_factor/prime_factor.js]: Function Docstring missing.

Comment (Prakash2403) [inline prime_factor/prime_factor.js]: Confine i to sqrt(n)

Comment (aashutoshrathi) [inline prime_factor/prime_factor.js]: We should check until sqrt(n).

**Pull request merged**